csrc/scheduler/normalization_inner_outer_utils.cpp

```diff
@@ -221,20 +240,26 @@ PersistentBufferStorageParams getPersistentBufferStorageParams(
   for (auto buffer : buffers) {
     int64_t buffer_size_regs = scheduler_utils::getPersistentBufferSizeOfTensor(
         buffer, runtime_info, persistent_buffer_info);
-    int64_t buffer_size_smem = roundUpSharedMemory(
-        buffer_size_regs,
-        dataTypeSize(buffer->getDataType().value()),
-        vectorize_factor,
-        threads_per_block_min,
-        threads_per_block_max,
-        dev_prop->warpSize);
+    // When warp specialized, the whole buffer is loaded in a single TMA
+    // instruction. No round up issue due to non-divisible split.
+    int64_t buffer_size_smem = is_warp_specialized
+        ? buffer_size_regs
+        : roundUpSharedMemory(
+              buffer_size_regs,
+              dataTypeSize(buffer->getDataType().value()),
+              vectorize_factor,
+              threads_per_block_min,
+              threads_per_block_max,
+              dev_prop->warpSize);
     required_size_regs_smem_map[buffer] =
         std::make_pair(buffer_size_regs, buffer_size_smem);
     total_smem_buffer_size += buffer_size_smem;
   }
   buffer_params.smem_buffer_size = total_smem_buffer_size;
-  buffer_params.regs_buffer_size =
+  buffer_params.regs_buffer_size +=
```

Comment:
is this intended? `buffer_params.regs_buffer_size` is initialized as -1. So the math isn't right after this change?